src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/RSACryptoServiceProvider.cs

```diff
@@ -132,17 +206,7 @@ public bool PersistKeyInCsp
         {
             get
             {
-                if (_safeProvHandle == null)
-                {
-                    lock (this)
-                    {
-                        if (_safeProvHandle == null)
-                        {
-                            _safeProvHandle = CapiHelper.CreateProvHandle(_parameters, _randomKeyContainer);
-                        }
-                    }
-                }
-                return CapiHelper.GetPersistKeyInCsp(_safeProvHandle);
+                return CapiHelper.GetPersistKeyInCsp(SafeProvHandle);
```

Comment:
I don't think this is thread safe against the provHandle getting replaced. That said, this type isn't documented as thread safe, so why do we have all of this locking?
Arguing with myself here, I'm a little concerned about the possibility of memory safety issues if this code isn't made threadsafe